WooCommerce/Classes/Authentication/Store Creation/Progress/StoreCreationProgressViewModel.swift

```diff
@@ -0,0 +1,165 @@
+    }
+
+    private var animationTimer: Timer?
+    private let progressViewAnimationTimerInterval: TimeInterval
+    private let incrementProgressValueBy: Float
+
+    /// - Parameters:
+    ///   - incrementInterval: Interval at which progress will be incremented to next case
+    ///   - progressViewAnimationTimerInterval: Animation timer interval DI for unit test purposes.
+    init(incrementInterval: TimeInterval,
+         progressViewAnimationTimerInterval: TimeInterval = 0.1) {
+        self.progressViewAnimationTimerInterval = progressViewAnimationTimerInterval
+        // Increment the progress value until next progress increment
+        let gapBetweenProgress = Progress.allCases[1].rawValue - Progress.allCases[0].rawValue
+        self.incrementProgressValueBy = (gapBetweenProgress / Float(incrementInterval / progressViewAnimationTimerInterval))
+    }
+
+    // MARK: Public methods
+    //
+    func onAppear() {
+        animationTimer = Timer.scheduledTimer(withTimeInterval: progressViewAnimationTimerInterval, repeats: true) { [weak self] _ in
+            self?.animateProgressView()
+        }
+    }
+
+    func incrementProgress() {
+        guard let next = progress.next() else {
+            return
+        }
+        self.progress = next
+    }
+
+    func markAsComplete() {
+        self.progress = .finished
+    }
+}
+
+private extension StoreCreationProgressViewModel {
+    func animateProgressView() {
+        guard let next = progress.next(),
+              progressValue < next.rawValue else {
+            return
+        }
+        progressValue = min(progressValue + incrementProgressValueBy, next.rawValue)
+    }
+
+    enum Localization {
+        static let funWooFact = NSLocalizedString("#FunWooFact:", comment: "Prefix for the subtitle text in the store creation loading screen")
+    }
+}
+
+private extension StoreCreationProgressViewModel.Progress {
+    var title: String {
+        switch self {
+        case .creatingStore:
+            return Localization.Title.wooWeAreCreatingYourStore
+        case .buildingFoundations:
+            return Localization.Title.buildingTheFoundations
+        case .organizingStockRoom:
+            return Localization.Title.organizingTheStockRoom
+        case .applyingFinishingTouches, .finished:
+            return Localization.Title.finishingTouches
+        }
+    }
+
+    var subtitle: String {
+        switch self {
+        case .creatingStore:
+            return Localization.Subtitle.stores
+        case .buildingFoundations:
+            return Localization.Subtitle.founded
+        case .organizingStockRoom:
+            return Localization.Subtitle.catOrDog
+        case .applyingFinishingTouches, .finished:
+            return Localization.Subtitle.meetups
+        }
+    }
+}
+
+private extension StoreCreationProgressViewModel.Progress {
+    func next() -> Self? {
+        let all = Self.allCases
+        guard let idx = all.firstIndex(of: self) else {
```

Comment:
nit: I think we generally use the full name especially when the variable isn't too long.
```suggestion
        guard let index = all.firstIndex(of: self) else {
```

Reply:
Done as part of f07b473ecfceea9cea553ffd80c7f403fdac5bfb

Reply:
thanks for adding unit tests for this!